tests/run-pass/concurrency/libc_pthread_cond.rs

```diff
@@ -0,0 +1,199 @@
+        let cond = Arc::try_unwrap(cond).unwrap();
+        assert_eq!(libc::pthread_cond_destroy(cond.inner.get() as *mut _), 0);
+    }
+}
+
+fn test_broadcast() {
+    unsafe {
+        let cond = Arc::new(create_cond(None));
+        let mutex = Arc::new(create_mutex());
+
+        assert_eq!(libc::pthread_mutex_lock(mutex.inner.get() as *mut _), 0);
+
+        let spawn_mutex = Arc::clone(&mutex);
+        let spawn_cond = Arc::clone(&cond);
+        let handle = thread::spawn(move || {
+            assert_eq!(libc::pthread_mutex_lock(spawn_mutex.inner.get() as *mut _), 0);
+            assert_eq!(libc::pthread_cond_broadcast(spawn_cond.inner.get() as *mut _), 0);
+            assert_eq!(libc::pthread_mutex_unlock(spawn_mutex.inner.get() as *mut _), 0);
+        });
+
+        assert_eq!(
+            libc::pthread_cond_wait(cond.inner.get() as *mut _, mutex.inner.get() as *mut _),
+            0
+        );
+        assert_eq!(libc::pthread_mutex_unlock(mutex.inner.get() as *mut _), 0);
+
+        handle.join().unwrap();
+
+        let mutex = Arc::try_unwrap(mutex).unwrap();
+        assert_eq!(libc::pthread_mutex_destroy(mutex.inner.get() as *mut _), 0);
+        let cond = Arc::try_unwrap(cond).unwrap();
+        assert_eq!(libc::pthread_cond_destroy(cond.inner.get() as *mut _), 0);
+    }
+}
+
+fn test_timed_wait_timeout() {
+    unsafe {
+        let attr = create_cond_attr_monotonic();
+        let cond = create_cond(Some(attr));
+        let mutex = create_mutex();
+        let timeout = create_timeout(1);
+
+        assert_eq!(libc::pthread_mutex_lock(mutex.inner.get() as *mut _), 0);
+        assert_eq!(
+            libc::pthread_cond_timedwait(
+                cond.inner.get() as *mut _,
+                mutex.inner.get() as *mut _,
+                &timeout
+            ),
+            libc::ETIMEDOUT
+        );
+        assert_eq!(libc::pthread_mutex_unlock(mutex.inner.get() as *mut _), 0);
+        assert_eq!(libc::pthread_mutex_destroy(mutex.inner.get() as *mut _), 0);
+        assert_eq!(libc::pthread_cond_destroy(cond.inner.get() as *mut _), 0);
+    }
+}
+
+fn test_timed_wait_notimeout() {
+    unsafe {
+        let attr = create_cond_attr_monotonic();
+        let cond = Arc::new(create_cond(Some(attr)));
+        let mutex = Arc::new(create_mutex());
+        let timeout = create_timeout(100);
+
+        assert_eq!(libc::pthread_mutex_lock(mutex.inner.get() as *mut _), 0);
+
+        let spawn_mutex = Arc::clone(&mutex);
+        let spawn_cond = Arc::clone(&cond);
+        let handle = thread::spawn(move || {
+            assert_eq!(libc::pthread_mutex_lock(spawn_mutex.inner.get() as *mut _), 0);
+            assert_eq!(libc::pthread_cond_signal(spawn_cond.inner.get() as *mut _), 0);
+            assert_eq!(libc::pthread_mutex_unlock(spawn_mutex.inner.get() as *mut _), 0);
+        });
+
+        assert_eq!(
+            libc::pthread_cond_timedwait(
```

Comment:
Is there no way to trigger this through libstd?
Also, could we measure the time that we are actually waiting to make sure it is roughly like what we would expect?